### PR TITLE
[ALLUXIO-1755] Manage BlockMasterClient via try-with-resources

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-1755

### CODE REVIEW

File: core/common/src/main/java/alluxio/resource/CloseableResource.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  /**
+   * @return the resource
+   */
+  public T get() {
+    return mResource;
+  }
+
+  /**
+   * Performs any cleanup operations necessary when the resource is no longer in use.
+   */
```
Comment (yupeng9) [inline core/common/src/main/java/alluxio/resource/CloseableResource.java]: @Ovverride

**Comment by yupeng9** [comment]
Excellent improvement! Thanks.

**Comment by yupeng9** [comment]
LGTM

File: core/common/src/main/java/alluxio/resource/CloseableResource.java (hunk 1 above)
Comment (calvinjia) [inline core/common/src/main/java/alluxio/resource/CloseableResource.java]: Has this comment been addressed?

Comment (aaudiber) [inline core/common/src/main/java/alluxio/resource/CloseableResource.java]: Addressed

**Comment by apc999** [comment]
LGTM

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**